microcosm_eventsource/immutable_event_factory.py

```diff
@@ -0,0 +1,63 @@
+"""
+Factory for immutable events.
+
+In extension of EventFactory, this class provides a native support of event updating container objects based
+on event types
+
+"""
+from collections import defaultdict
+
+container_event_mutator_registry = defaultdict(list)
+
+
+def register_container_mutator_common(event_type):
+    def decorator(func):
+        container_event_mutator_registry[event_type.__name__].append(func.__name__)
+        return func
+
+    return decorator
+
+
+def register_container_mutator_by_event_type(event_instance_type):
+    def decorator(func):
+        container_event_mutator_registry[event_instance_type.name].append(func.__name__)
+        return func
+
+    return decorator
```

Comment:
Seems like those two decorators are basically doing the same thing, but with different args. Could we unify into one? I'm rather in favor of short decorator names (`register_container_mutator` in this case).

Reply:
I am effectively using same registry for two different handlers (common handlers and event specific handler). Let me split those out to have a clear defined roles.

Reply:
created separate registry.